howto/android/odm_vhal.md

```diff
@@ -0,0 +1,128 @@
+*since 1.22.0*
+
+[note type="information" status="Note"]Replacing the Anbox VHAL is only supported on AAOS images.
+The Anbox Cloud dashboard does not support custom VHAL implementations.[/note]
+
+This document will guide through the process of replacing the Anbox Cloud VHAL
+implementation with your own implementation placed in the
+[ODM partition](https://source.android.com/docs/core/architecture/partitions/odm-partitions)
+using an [addon](https://anbox-cloud.io/docs/explanation/addons).
+
+Your custom VHAL implementation must provide an
+[HIDL VHAL interface](https://source.android.com/docs/automotive/vhal/hidl-vhal-interface)
+and be built against VNDK 29.
+
+## Create the addon
+
+This addon will be tasked with the following:
+
+1. Copying the VHAL binary to `/odm/bin/hw`
+1. Copying an
+[init file](https://android.googlesource.com/platform/system/core/+/master/init/README.md)
+for your VHAL service to `/odm/etc/init`
+1. Disabling the Anbox Cloud VHAL by setting the `ro.anbox.automotive.vhal`
+property to `odm`. This will be done by writing that system property to the
+`/odm/etc/build.prop` file.
+
+In the following, we will name our addon `custom-vhal`.
+To override the VHAL implementation, we will use a
+[pre-start hook](https://anbox-cloud.io/docs/reference/hooks) which is executed
+before Android gets started.
+
+The directory layout for the addon is strict and must be:
+
+```
+custom-vhal/
+├── hooks
+│   └── pre-start
+├── manifest.yaml
+├── vhal
+└── vhal.rc
+```
+
+`vhal` is the VHAL implementation built in the previous section. If your VHAL
+implementation requires additional libraries, they must also be bundled with the
+addon and copied in the ODM partition.
+
```

Comment:
The following can also be part of the prerequisites section, the idea is to have everything the user needs to do *before* the actual steps of a how-to guide in the prerequisites/preparation section.

" If your VHAL implementation requires additional libraries, they must also be bundled with the
addon and copied in the ODM partition."

Reply:
Makes sense. I will add a small reminder here to ensure people remember.

Reply:
Please tell me if it's better now. I can also remove the small reminder if you think it's redundant.

Reply:
Looks good now.